src/utils.coffee

```diff
@@ -102,27 +102,28 @@ exports.connectivityCheck = _.once ->
 				blink.pattern.start(networkPattern)
 
 
-apiSecretPromise = null
-generateApiSecret = ->
+secretPromises = {}
+generateSecret = (name) ->
 	Promise.try ->
-		return config.forceApiSecret ? randomHexString.generate()
+		return config.forceApiSecret if name == 'apiSecret' && config.forceApiSecret?
```

Comment:
You made the function generic, but this if in here feels dirty. A separate specific function for `'apiSecret'` a way to resolve this.